### PR TITLE
Adds appconfig, core, and eventgrid to PR pipelines.

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR should get merged after #6101. It sets up the CI pipelines for:

* sdk/appconfiguration
* sdk/core
* sdk/eventgrid

### CODE REVIEW

File: sdk/core/ci.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# DO NOT EDIT THIS FILE
+# This file is generated automatically and any changes will be lost.
+
+pr:
+  branches:
+    include:
+    - master
+  paths:
+    include:
+    - sdk/core/
```
Comment (pakrym) [inline sdk/core/ci.yml]: when core changes all dependent projects should be rebuilt.

cc @weshaggard

Comment (weshaggard) [reply sdk/core/ci.yml]: Agreed. @mitch we should point this to eng/data-plane.proj.

Comment (mitchdenny) [reply sdk/core/ci.yml]: Updated.

**Comment by mitchdenny** [comment]
Added one more change to this. Turns out that these new pipelines were triggering as CI pipelines on every commit. I've disabled the default behavior there. When we create the tool to automatically generate the pipelines we might change this so its an option on the pipeline definition, not the YAML.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**